### ISSUE TITLE
The url with file extension wasn't matched now

### ISSUE THREAD

**Issue by kexxxfeng** [aspnet/Docs#6769]
The route template:
```
routes.MapRoute("IndexPage",
                "feed/{page}.html",
                new { controller = "User", action = "Index" }
            );
```
the request url: `http://www.abc.com/feed/2.html`

It works in aspnetcore 2.0, but doesn't work after the project upgraded to 2.1.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 32374077-e02e-d6a6-1cea-05e58447b7cd
* Version Independent ID: 86aeab13-1c25-9fc1-4e56-0390cb40c242
* Content: [Routing in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1#feedback)
* Content Source: [aspnetcore/fundamentals/routing.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/routing.md)
* Service: **unspecified**
* Product: **asp.net-core**
* GitHub Login: @ardalis
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
@kexxxfeng  have you tried https://docs.microsoft.com/en-us/aspnet/core/fundamentals/startup?view=aspnetcore-2.1#setcompatibilityversion-for-aspnet-core-mvc ?

**Comment by Rick-Anderson** [comment]
@kexxxfeng  where do you see a problem on [Routing in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1#feedback) ?

**Comment by kexxxfeng** [comment]
@Rick-Anderson Yes, I have `.SetCompatibilityVersion(CompatibilityVersion.Version_2_1)`, it doesn't work.
I followed [this section](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-2.1#route-template-reference) that referenced `file extension`, and the code works well in aspnet core 2.0.

**Comment by Rick-Anderson** [comment]
You'll get faster response debugging your code if you post to StackOverflow. We're really not in a position to debug code here.
